compiler/src/lnntoamm/Types.ts

```diff
@@ -156,22 +173,61 @@ export class Builtin extends Type {
     }
   }
 
+  fieldIndices(): FieldIndices {
+    return TODO("determine if it's even worth keeping the Builtin class");
+  }
+
   instance(): Type {
     return this;
   }
 
+  isFixed(): boolean {
+    switch (this.name) {
+      case 'int64':
+      case 'int32':
+      case 'int16':
+      case 'int8':
+      case 'float64':
+      case 'float32':
+      case 'bool':
+      case 'void':
+        return true;
+      default:
+        return false;
+    }
+  }
+
   tempConstrain(ty: Type, scope: Scope) {
     this.constrain(ty, scope);
   }
 
   resetTemp() {
     // do nothing
   }
+
+  size(): number {
+    // TODO: should probably figure out non-opaque types but we'll see
+    // what happens with this class first (after fn selection fix)
+    switch (this.name) {
+      case 'void':
+        return 0;
+      default:
+        // FIXME: currently the AVM encodes every value (including 8-bit values)
+        // in 8 bytes, so we have to enforce that behavior here by assuming that
+        // the size is in factors of 8 bytes. Eventually, we'll have to fix this.
+        return 1;
```

Comment:
Or we just pretend 64-bits is sufficient for everything and don't. ;)

Reply:
true! will update comment